tools/cli_commands/test/test_gpg_encrypt.py

```diff
@@ -53,7 +57,7 @@ def test_gpg_encrypt_from_vault(get_users_by_mock, gpg_encrypt_mock):
 
     command.execute()
 
-    secret_reader_mock.assert_called_once_with({"path": vault_secret_path})
+    secret_reader_mock.assert_called_once_with({"path": vault_secret_path})  # type: ignore[attr-defined]
```

Comment:
one way to avoid ignore typing is to extract a `build_secret_reader_mock(secret)` and pass result to `craft_command` instead of build `secret_reader` inside it, then use mock object directly in test cases shouldn't cause typing issues.

Reply:
Changed in [f23fd92](https://github.com/app-sre/qontract-reconcile/pull/4839/commits/f23fd926cc8927e62a13b604ba6f78f7251e3cec)

Reply:
actually no need to patch `reconcile.utils.secret_reader.SecretReader` as it's used as dependency when init `GPGEncryptCommand`, and in each test case, `secret` is set as `{"x": "y"}`, 

```python
SECRET = {"x": "y"}


@pytest.fixture
def secret_reader() -> MagicMock:
    mock = create_autospec(SecretReader)
    mock.read_all.return_value = SECRET
    return mock
```

also need to ensure assert on `secret_reader.read_all` instead of just `secret_reader`

Reply:
Done in [066fc0b](https://github.com/app-sre/qontract-reconcile/pull/4839/commits/066fc0bc45be7a6432d2556b2468f918cb9d69e5)